### PR TITLE
Round 9

### PR DESCRIPTION
**Pull request by jasnell**
/cc @mcollina 

Here we go....

### CODE REVIEW

File: test/parallel/test-http2-server-startup.js [hunk 1]
```diff
@@ -6,11 +6,11 @@
```
Comment (thefourtheye) [inline test/parallel/test-http2-server-startup.js:6]: File permissions are changed.

**Review by mcollina — APPROVED** [review]
LGTM

File: lib/internal/http2/util.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  isIllegalConnectionSpecificHeader,
+  llistToHeaders,
+  mapToHeaders
+};
+
+
+
+/*
+onStream stuff
+  var request =
```
Comment (YurySolovyov) [inline lib/internal/http2/util.js:73]: is that comment really needed?

Comment (jasnell) [reply lib/internal/http2/util.js:73]: This is just temporary. Keeping around some of the refactored out code so it's not lost as we shift a few other bits around.

**Comment by italoacasas** [comment]
@jasnell do you think make sense to assert that the fallowing function are there ?
```js
- createServer()
- createSecureServer()
- createServerSession()
- createClientSession()
```

**Comment by jasnell** [comment]
Yeah, I'm thinking we should assert the presence of all the primary APIs. The tests here only barely scratch the surface.

**Comment by jasnell** [comment]
Quite a few additional updates here. For the most part this is pretty solid. Push is now enabled.

@mcollina: There are limits in here. In a load benchmark throwing 1,000,000 requests at once, most requests fail and eventually the server will hit a segfault attempting to free some memory. I haven't yet tracked down where that segfault is happening.

**Comment by jasnell** [comment]
@mcollina ... I'm going to get this round landed. The current implementation has no apparently leaks and the segfault appears to be closed (at least in my most recent tests).

**Comment by jasnell** [comment]
Landed... will start working on round10

**Pull request closed without merging**